src/DateRangePicker.ts

```diff
@@ -1,26 +1,423 @@
+      border-radius: 3px;
+      padding: 2px;
+      font-size: ${tooltipFontSize};
+      font-family: sans-serif;
+      touch-action: none;
+      pointer-events: none;
+    }
+    #tooltip:after {
+      content: '';
+      position: absolute;
+      margin-left: -5px;
+      top: 100%;
+      /* arrow */
+      border: 5px solid ${tooltipTextColor};
+      border-color: ${tooltipBackgroundColor} transparent transparent
+        transparent;
+    }
+    /****** slider ********/
+    .draggable:hover {
+      cursor: grab;
+    }
+    .dragging {
+      cursor: grabbing !important;
+    }
+    /****** inputs ********/
+    #inputs {
+      display: flex;
+      justify-content: center;
+    }
+    #inputs .dash {
+      position: relative;
+      bottom: 1px;
+    }
+    input {
+      width: ${inputWidth};
+      margin: 0 3px;
+      border: ${inputBorder};
+      border-radius: 2px !important;
+      text-align: center;
+      font-size: ${inputFontSize};
     }
   `;
 
-  @property({ type: String }) title = 'Hey there';
+  firstUpdated() {
+    this._leftSliderX = this.sliderWidth;
+    this._rightSliderX = this.width - this.sliderWidth;
+    this._minDate = dayjs(this.data?.minDate).valueOf();
+    this._maxDate = dayjs(this.data?.maxDate).valueOf();
+    this._histWidth = this.width - this.sliderWidth * 2;
+    this._numBins = this.data?.bins?.length ?? 1;
+    this._binWidth = this._histWidth / this._numBins;
+    this._histData = this.generateHistData();
+  }
+
+  generateHistData(): HistogramItem[] {
+    if (!this.data) {
+      return [];
+    }
+    const minValue = Math.min(...this.data.bins);
+    const maxValue = Math.max(...this.data.bins);
+    const valueScale = this.height / Math.log1p(maxValue - minValue);
+    const dateScale = this.dateRange() / this._numBins;
+    return this.data.bins.map((v: number, i: number) => {
+      return {
+        value: v,
+        height: Math.floor(Math.log1p(v) * valueScale),
+        binStart: `${dayjs(i * dateScale + this._minDate).format(
+          this.dateFormat
+        )}`,
+        binEnd: `${dayjs((i + 1) * dateScale + this._minDate).format(
+          this.dateFormat
+        )}`,
+      };
+    });
+  }
+
+  dateRange(): number {
+    return this._maxDate - this._minDate;
+  }
 
-  @property({ type: Number }) counter = 5;
+  @query('#tooltip') tooltip!: HTMLDivElement;
 
-  __increment() {
-    this.counter += 1;
+  showTooltip(e: PointerEvent): void {
+    const target = e.currentTarget as SVGRectElement;
+    const x = target.x.baseVal.value + this.sliderWidth / 2;
+    const data = target.dataset;
+    const itemsText = `item${data.numItems !== '1' ? 's' : ''}`;
+    const tooltipOffset =
+      x + (this._binWidth - this.sliderWidth - this.tooltipWidth) / 2;
+    this.tooltip.style.left = `${tooltipOffset}px`;
+    this.tooltip.innerHTML = `${data.numItems} ${itemsText}<br>${data.binStart} - ${data.binEnd}`;
+    this.tooltip.style.display = 'block';
```

Comment:
For these three lines, I would recommend using reactive properties instead of setting the style / content directly on the element. ie:
```
this.tooltipLeft = tooltipOffset;
this.tooltipContent = `${data.numItems} ${itemsText}<br>${data.binStart} - ${data.binEnd}`;
this.tooltipVisible = true;
...
render html`
  <div 
    id="tooltip" 
    class="${this.tooltipVisible ? '' : 'hidden"}" 
    style="left: ${this.tooltipLeft}px"
  >
    ${this.tooltipContent}
  </div>
`
```